app/controllers/PaypalController.scala

```diff
@@ -88,87 +98,10 @@ class PaypalController(ws: WSClient, paymentServices: PaymentServices, checkToke
-      *
-      * See: https://www.w3.org/TR/referrer-policy/#strip-url
-      *
-      */
-    def withSafeRefererUrl(
-                            countryGroup: CountryGroup,
-                            amount: BigDecimal,
-                            cmp: Option[String],
-                            intCmp: Option[String],
-                            refererPageviewId: Option[String],
-                            refererUrl: Option[String],
-                            ophanPageviewId: Option[String],
-                            ophanBrowserId: Option[String],
-                            ophanVisitId: Option[String]
-                          ): AuthRequest = {
-      val safeRefererUrl = refererUrl.flatMap(url => Try(Uri.parse(url).copy(fragment = None).toString).toOption)
-
-      new AuthRequest(countryGroup, amount, cmp, intCmp, refererPageviewId, safeRefererUrl, ophanPageviewId, ophanBrowserId, ophanVisitId)
-    }
-
-
-    implicit val authRequestReads: Reads[AuthRequest] = (
-      (__ \ "countryGroup").read[CountryGroup] and
-        (__ \ "amount").read(min[BigDecimal](1)) and
-        (__ \ "cmp").readNullable[String] and
-        (__ \ "intCmp").readNullable[String] and
-        (__ \ "refererPageviewId").readNullable[String] and
-        (__ \ "refererUrl").readNullable[String] and
-        (__ \ "ophanPageviewId").readNullable[String] and
-        (__ \ "ophanBrowserId").readNullable[String] and
-        (__ \ "ophanVisitId").readNullable[String]
-      ) (AuthRequest.withSafeRefererUrl _)
-  }
-
-  case class AuthResponse(approvalUrl: Uri, paymentId: String)
-
-  object AuthResponse {
-    import cats.syntax.either._
-    import scala.collection.JavaConverters._
-
-    def fromPayment(payment: Payment): Either[String, AuthResponse] = Either.fromOption(for {
-        links <- Option(payment.getLinks)
-        approvalLinks <- links.asScala.find(_.getRel.equalsIgnoreCase("approval_url"))
-        approvalUrl <- Option(approvalLinks.getHref)
-        paymentId <- Option(payment.getId)
-      } yield AuthResponse(Uri.parse(approvalUrl), paymentId), "Unable to parse payment")
-  }
-
-  implicit val UriWrites = new Writes[Uri] {
-    override def writes(uri: Uri): JsValue = JsString(uri.toString)
-  }
-
-  implicit val AuthResponseWrites = Json.writes[AuthResponse]
-
-  implicit val CountryGroupReads = new Reads[CountryGroup] {
-    override def reads(json: JsValue): JsResult[CountryGroup] = json match {
-      case JsString(id) => CountryGroup.byId(id).map(JsSuccess(_)).getOrElse(JsError("invalid CountryGroup id"))
-      case _ => JsError("invalid value for country group")
-    }
-  }
-
-  private def capAmount(amount: BigDecimal, currency: Currency): BigDecimal = amount min MaxAmount.forCurrency(currency)
-
   def authorize = checkToken {
     NoCacheAction.async(parse.json[AuthRequest]) { implicit request =>
+      info(s"Attempting to obtain paypal auth url.")
```

Comment:
_attempting to create Paypal payment_ may be clearer - the payment is then used to derive the approval url and payment id